polyfill/lib/ecmascript.mjs

```diff
@@ -571,7 +571,9 @@ export const ES = ObjectAssign({}, ES2020, {
     ES.RejectDuration(years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds);
     for (const property of disallowedProperties) {
       if (record[property] !== 0) {
-        throw new RangeError(`invalid duration field ${property}`);
+        throw new RangeError(
+          `Duration field ${property} not supported by Temporal.Instant. Try Temporal.ZonedDateTime instead.`
```

Comment:
Even though this Abstract Operation is generic in theory, in practice it's only used by the Instant type (and we're not changing the API anymore!) so this type-specific error message seems safe.

Reply:
@justingrant also since the error messages aren't specified, I think it's not a problem :)